### PR TITLE
[BUG] Fix computation of welch's PSD when the number of jobs exceeds the number of channels

### LINKED ISSUE #11297: compute_psd: ValueError when n_jobs=-1 and only 1 channel
### Description of the problem

when [`compute_psd`](https://mne.tools/stable/generated/mne.io.Raw.html) is passed `n_jobs=-1` and the instance contains only _one_ channel, the method fails.

### Steps to reproduce

```Python
import numpy as np
import mne
data = np.random.randn(1, 4096) 
info = mne.create_info(ch_names=["EEG 01"], ch_types="eeg")
raw = mne.io.RawArray(data, info)
spectrum = raw.compute_psd(method="welch", n_jobs=-1)
```
```


### Link to data

_No response_

### Expected results

no error

### Actual results

```
ValueError: all the input arrays must have same number of dimensions, but the array at index 0 has 2 dimension(s) and the array at index 1 has 1 dimension(s)
```

### Additional information

Platform:         Linux-5.19.16-76051916-generic-x86_64-with-glibc2.35
Python:           3.10.6 (main, Aug 10 2022, 11:40:04) [GCC 11.3.0]
Executable:       /usr/bin/python3
CPU:              x86_64: 8 cores
Memory:           15.3 GB

mne:              1.2.1
numpy:            1.23.4 {OpenBLAS 0.3.20 with 8 threads}
scipy:            1.9.3
matplotlib:       3.6.0 {backend=module://matplotlib_inline.backend_inline}

sklearn:          1.1.3
numba:            Not found
nibabel:          Not found
nilearn:          Not found
dipy:             Not found
openmeeg:         Not found
cupy:             Not found
pandas:           1.5.1
pyvista:          Not found
pyvistaqt:        Not found
ipyvtklink:       Not found
vtk:              Not found
qtpy:             2.2.1 {PyQt5=5.15.3}
ipympl:           Not found
pyqtgraph:        Not found
pooch:            v1.6.0

mne_bids:         Not found
mne_nirs:         Not found
mne_features:     Not found
mne_qt_browser:   Not found
mne_connectivity: Not found
mne_icalabel:     Not found

### PR DESCRIPTION
**Pull request by mscheltienne**
Closes #11297 
cc @mmagnuski

### CODE REVIEW

File: mne/time_frequency/tests/test_psd.py [hunk 1]
```diff
@@ -192,3 +192,10 @@ def test_compares_psd():
     assert (np.sum(freqs_scipy < 0) == 0)
     assert (np.sum(psds_mne < 0) == 0)
     assert (np.sum(psds_scipy < 0) == 0)
+
+
+def test_psd_array_welch_n_jobs():
+    """Test that n_jobs works even with more jobs than channels."""
+    data = np.random.randn(1, 2048)
```
Comment (agramfort) [inline mne/time_frequency/tests/test_psd.py:199]:
```suggestion
    data = np.empty(1, 2048)
```

no need to generate numbers here.

**Review by agramfort — APPROVED** [review]
just a nitpick. thx @mscheltienne

**Review by drammock — APPROVED** [review]

File: mne/time_frequency/tests/test_psd.py [hunk 2]
```diff
@@ -192,3 +192,10 @@ def test_compares_psd():
     assert (np.sum(freqs_scipy < 0) == 0)
     assert (np.sum(psds_mne < 0) == 0)
     assert (np.sum(psds_scipy < 0) == 0)
+
+
+def test_psd_array_welch_n_jobs():
+    """Test that n_jobs works even with more jobs than channels."""
+    data = np.empty(1, 2048)
```
Comment (mscheltienne) [inline mne/time_frequency/tests/test_psd.py:199]:
```suggestion
    data = np.empty((1, 2048))
```

**Comment by sam-s** [comment]
please add a test with `n_jobs=-1`
thank you.

**Comment by mscheltienne** [comment]
@sam-s `n_jobs=-1` will spawn as many workers as there are cores, which on CI is probably 2 anyway (depends on the CI provider).

And to explain the issue and the fix, the input to the // function is split on the channel axis by the number of workers. For instance, if you have 4 channels and 2 workers, both will receive 2 channels. If you have 4 channels and 4 workers, both will receive one channel. And now for the breaking edge-case, if you have more workers than channels, some workers were receiving arrays of shape `(0, n_times)`, i.e. 0 channels, and were put to work on those empty arrays, thus returning nothing and raising during the concatenation of results. 
The fix is thus very simple, you don't use more workers than channels even if the user provides an invalid number of jobs for the provided array.

**Comment by sam-s** [comment]
> @sam-s `n_jobs=-1` will spawn as many workers as there are cores, which on CI is probably 2 anyway (depends on the CI provider).

I know, the issue is when `-1` is replaced by the actual number of cores - I was afraid that this error might happen before the replacement (and thus you get something like `4 // -1` &c)
I suppose you thought about it already.
Thank you for fixing the issue so fast.

**Comment by mscheltienne** [comment]
No risk on that side, arguments like `n_jobs` are checked and converted to a value at the beginning of the function, and I did check locally with `-1` anyway ;) Do note however that providing a number of jobs above the number of channels will not yield any performance benefit.

**Pull request merged**